network/wsPeer.go

```diff
@@ -368,6 +374,8 @@ func (wp *wsPeer) writeLoop() {
 				return
 			}
 			continue
+		case <-wp.closing:
+			return
```

Comment:
This was another place where having just a single case with a default was somewhat faster than having two cases, and I believe it's equally correct to omit this `<-wp.closing` case here.

Reply:
I think that we don't want to send anything once we have the closing channel closed.
( we don't want to drain the channel slowly; instead we want to just ignore it completely )

What if we would drain the sendBufferHighPrio channel in the Close() function before CloseWithoutFlush ?

Reply:
or add an atomic test for didInnerClose within writeLoopSend ?

Reply:
After a `wsPeer` is closed, subsequent calls to `wp.writeLoopSend` will return true, which will cause the above case statement to return.  So, I don't think an additional case is necessary to cause the `writeLoop` goroutine to exit promptly on peer closure.

Reply:
Yes, I missed the return value; it's ok then. As long as we're not iterating on all the pending requests that we don't care about anyway, we can skip the closing case here.